File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        
        temp = dict()
        
        for num in nums:
            if num in temp:
                return True
            temp[num] = num
        
        return False
        

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
     
        HashSet<Integer> map = new HashSet<>();
        
        for(int num : nums){
            if(map.contains(num)){
                return true;
            }else{
                map.add(num);
            }
        }
        
        return false;
    }
}